labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise02.scala

```diff
@@ -39,29 +39,34 @@ object ListManipulationExercise02 {
    * in a one-liner.
    */
   def separateTheYoungFromTheOld(persons: List[Person]): List[List[String]] = {
-    var youngins: ListBuffer[Person] = new ListBuffer[Person]()
-    var elders: ListBuffer[Person] = new ListBuffer[Person]()
-    var validYoungNames: ListBuffer[String] = new ListBuffer[String]()
-    var validOldNames: ListBuffer[String] = new ListBuffer[String]()
-
-    for (person <- persons) {
-        if (person.age < 18) {
-          youngins += person
-        } else {
-          elders += person
-        }
-    }
-
-    var sortedYoung = youngins.toList.sortBy(_.age)
-    var sortedOld = elders.toList.sortBy(_.age)
-
-    for (young <- sortedYoung) {
-      validYoungNames += young.firstName
-    }
-    for (old <- sortedOld) {
-      validOldNames += old.firstName
-    }
-    List(validYoungNames.toList, validOldNames.toList)
+//    var youngins: ListBuffer[Person] = new ListBuffer[Person]()
+//    var elders: ListBuffer[Person] = new ListBuffer[Person]()
+//    var validYoungNames: ListBuffer[String] = new ListBuffer[String]()
+//    var validOldNames: ListBuffer[String] = new ListBuffer[String]()
+//
+//    for (person <- persons) {
+//        if (person.age < 18) {
+//          youngins += person
+//        } else {
+//          elders += person
+//        }
+//    }
+//
+//    var sortedYoung = youngins.toList.sortBy(_.age)
+//    var sortedOld = elders.toList.sortBy(_.age)
+//
+//    for (young <- sortedYoung) {
+//      validYoungNames += young.firstName
+//    }
+//    for (old <- sortedOld) {
+//      validOldNames += old.firstName
+//    }
+//    List(validYoungNames.toList, validOldNames.toList)
+//  
+    val (youngins,elders) = persons partition (_.age < 18)
+    val youngNames = youngins.sortBy(_ age).map(_ firstName)
+    val oldNames = elders.sortBy(_ age).map(_ firstName)
+    List(youngNames, oldNames)
```

Comment:
I like the use of partition to get two lists in a tuple! If you want to see how to do it in one line, check what Josh Kutsko or I did :)